pyomo/contrib/gdpopt/loa.py

```diff
@@ -0,0 +1,281 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2022
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from collections import namedtuple
+from math import copysign
+
+from pyomo.common.collections import ComponentMap
+from pyomo.common.config import add_docstring_list
+from pyomo.common.modeling import unique_component_name
+from pyomo.contrib.gdpopt.algorithm_base_class import _GDPoptAlgorithm
+from pyomo.contrib.gdpopt.config_options import (
+    _add_OA_configs, _add_mip_solver_configs, _add_nlp_solver_configs,
+    _add_tolerance_configs)
+from pyomo.contrib.gdpopt.create_oa_subproblems import (
+    _get_master_and_subproblem, add_constraint_list)
+from pyomo.contrib.gdpopt.cut_generation import add_no_good_cut
+from pyomo.contrib.gdpopt.oa_algorithm_utils import _OAAlgorithmMixIn
+from pyomo.contrib.gdpopt.solve_master_problem import solve_MILP_master_problem
+from pyomo.contrib.gdpopt.util import (
+    time_code, _add_bigm_constraint_to_transformed_model)
+
+from pyomo.core import (
+    Block, Constraint, minimize, NonNegativeIntegers, NonNegativeReals,
+    Objective, value, Var, VarList)
+from pyomo.core.expr import differentiate
+from pyomo.core.expr.visitor import identify_variables
+from pyomo.gdp import Disjunct
+from pyomo.opt.base import SolverFactory
+from pyomo.repn import generate_standard_repn
+
+MAX_SYMBOLIC_DERIV_SIZE = 1000
+JacInfo = namedtuple('JacInfo', ['mode','vars','jac'])
+
+@SolverFactory.register(
+    'gdpopt.loa',
+    doc="The LOA (logic-based outer approximation) Generalized Disjunctive "
+    "Programming (GDP) solver")
+class GDP_LOA_Solver(_GDPoptAlgorithm, _OAAlgorithmMixIn):
+    """The GDPopt (Generalized Disjunctive Programming optimizer) logic-based
+    outer approximation (LOA) solver.
+
+    Accepts models that can include nonlinear, continuous variables and
+    constraints, as well as logical conditions. For nonconvex problems, LOA
+    may not report rigorous lower/upper bounds.
+    """
+    CONFIG = _GDPoptAlgorithm.CONFIG()
+    _add_OA_configs(CONFIG)
```

Comment:
This is just being picky, but all the other configs are lower case. Why is 'OA' not?

Reply:
Hehe, beats me!